style/js/shield_defs.js

```diff
@@ -243,6 +312,111 @@ export function loadShields(shieldImages) {
     },
   };
 
+  let padding_on_primary = {
+    left: 3,
+    right: 3,
+    top: 6,
+    bottom: 2,
+  };
+  shields["CA:ON:primary"] = {
+    backgroundImage: shieldImages.shield40_ca_on_primary,
+    textColor: "black",
+    padding: padding_on_primary,
+  };
+  shields["CA:ON:primary:Toll"] = {
+    backgroundImage: shieldImages.shield40_ca_on_primary_toll,
+    textColor: "white",
+    padding: padding_on_primary,
+  };
+  shields["CA:ON:private_toll"] = banneredShield(
+    roundedRectShield("white", "#003f87", "black", 8, 1, null),
+    ["ETR"]
+  );
+  shields["CA:ON:secondary"] = trapezoidDownShield;
+  shields["CA:ON:tertiary"] = roundedRectShield(
+    "white",
+    "black",
+    "black",
+    1,
+    1
+  );
+  shields["CA:ON:Halton"] = trapezoidUpShieldGreenYellow;
+  shields["CA:ON:Peel"] = trapezoidUpShieldBlackYellow;
+  shields["CA:ON:Simcoe"] = trapezoidUpShieldWhiteBlue;
+  ["Grey", "Hamilton", "Niagara"].forEach(
+    (county) => (shields[`CA:ON:${county}`] = trapezoidUpShieldBlueWhite)
+  );
+  [
+    "Brant",
+    "Bruce",
+    "Chatham-Kent",
+    "Cornwall",
+    "Dufferin",
+    "Durham",
+    "Elgin",
+    "Essex",
+    "Frontenac:Central Frontenac",
+    "Frontenac:Frontenac Islands",
+    "Frontenac:North Frontenac",
+    "Frontenac:South Frontenac",
+    "Greater Sudbury",
+    "Haldimand",
+    "Haliburton",
+    "Hastings:Carlow/Mayo",
+    "Hastings:Hastings Highlands",
+    "Hastings:Limerick",
+    "Hastings:Tyendinaga",
+    "Huron",
+    "Kawartha Lakes",
+    "Kingston",
+    "Lambton",
+    "Lanark",
+    "Leeds and Grenville",
+    "Lennox and Addington",
+    "Middlesex",
+    "Muskoka",
+    "Norfolk",
+    "Northumberland",
+    "Ottawa",
+    "Oxford",
+    "Perth",
+    "Peterborough",
+    "Prescott and Russell",
+    "Prince Edward",
+    "Quinte West",
+    "Renfrew",
+    "Stormont, Dundas and Glengarry",
+    "Waterloo",
+    "Wellington",
+    "York",
+  ].forEach(
+    (countyTownshipOrCity) =>
+      (shields[`CA:ON:${countyTownshipOrCity}`] = trapezoidUpShield)
+  );
+  shields["CA:ON:Hastings:Wollaston"] = banneredShield(
+    roundedRectShield("white", "black", "black", 1, 1),
+    ["TWP"]
+  );
+  shields["CA:ON:Waterloo:Wellesley"] = circleShield("white", "black");
+  shields["CA:ON:Waterloo:Woolwich"] = circleShield("white", "black");
+  ["North Dumfries", "Wilmot"].forEach(
+    (township) =>
+      (shields[`CA:ON:Waterloo:${township}`] = banneredShield(
+        shields["CA:ON:Waterloo"],
+        ["TWP"]
+      ))
+  );
+  ["Brant", "Durham", "Haldimand", "Norfolk"].forEach(
+    (county) =>
+      (shields[`CA:ON:${county}:Highway`] = banneredShield(
+        shields[`CA:ON:${county}`],
+        ["HWY"]
+      ))
+  );
+  shields["CA:ON:Muskoka:West"] = banneredShield(shields["CA:ON:Muskoka"], [
+    "WEST",
```

Comment:
The Muskoka Road 118 West shield is a bit awkward, since it reads backwards. We don’t have a way to put the banner below the shield, unfortunately. In the U.S., a route like this would be signposted with a “W” suffix that would easily fit in the `ref`, but it sounds like that isn’t typical in this region.

![](http://www.thekingshighway.ca/PHOTOS-3/hwy118-24_xlg.jpg)

Reply:
I thought Canada mostly had letter suffixes where the US has bannered routes. Nice to know there's at least one case where the opposite is true :upside_down_face: